internal/pkg/filesystem/filesystem.go

```diff
@@ -136,11 +137,30 @@ func IsAbs(v string) bool {
 
 // IsFrom returns true if path is from base dir or some sub-dir.
 func IsFrom(path, base string) bool {
+	path = strings.TrimRight(path, PathSeparatorString)
 	if base == "" || base == "." {
 		return true
 	}
-	baseWithSep := base + string(PathSeparator)
-	return strings.HasPrefix(path, baseWithSep)
+
+	lB := len(base)
+	lP := len(path)
+
+	// Path length must be greater than base length
+	if lP <= lB {
+		return false
+	}
+
+	// Path prefix must be equal to base
+	if path[0:lB] != base {
+		return false
+	}
+
+	// The prefix must be followed by the path separator
+	if path[lB] != PathSeparator {
+		return false
+	}
+
+	return true
```

Comment:
`base + ` ... alokuje novy string, ... spusta sa to pre kazdy objekt + `doUpdate` sa vola rekurzivne, takze je to velmi vela volani, ak je vela konfiguracii a config rows.

`TrimRight` iba oreze povodny string, takze sa nova pamat nealokuje, rovnako aj dalsie podmienky nealokuju novu pamat. Je to pokryte viacerými unit testami.

Reply:
Este lepsie by bolo negenerovat `paths` pre `remote` state, ale to bude vysledkom refactoringu.